_posts/2020-11-09-pcr-primer-calculator-using-r.md

```diff
@@ -0,0 +1,298 @@
+      - [No] calculate the minimum and maximum melting temperature
+  - create variables to hold the gc-percentage and annealing primer length
+  - create a data frame using the variables we created
+
+
+The bulk of our script will be wrapped around a For Loop. The code is written this way because we need to processes each line of sequences data at a time. Furthermore, we want to generate an output file for each sequences data we processed. The attributes of a For Loop fits perfectly for what we want to do.
+
+Once we are in the loop, we want to remove any spaces within the primer name and replace them with "_". Additionally we can format the sequences how we would like and then piece together the entire PCR primer
+```r
+# the For Loop will start on row 1. Once the For Loop cycles back to the top
+# of the loop, data on row 2 will be used and so forth
+# the For Loop ends once row_num excessed the value of num_of_rows_in_data
+for (row_num in 1:num_of_rows_in_data) {
+
+  # replace " " with "_" in the primer name
+  primer_name <- gsub(primer_information[row_num, 1],
+                      pattern = " ",
+                      replacement = "_")
+
+  # capitalize and/or lowercase the sequences using stringr.
+  # for appearance we want to make sure the sequence look uniform
+  # the restriction site sequences will be lowercase. This will help
+  # scientists to distinguish the leading strand and annealing sequence from
+  # the restriction site sequence.
+  annealing_seq <- stringr::str_to_upper(primer_information[row_num, 5])
+  restric_site <- stringr::str_to_lower(primer_information[row_num, 4])
+  lead_strand <- stringr::str_to_upper(primer_information[row_num, 3])
+
+  # combine the sequences together using paste0
+  whole_pcr_sequence <- paste0(lead_strand,
+                               restric_site,
+                               annealing_seq)
+
+  # in the output file we want to have 5' and 3' noted with the sequence.
+  # for ease, we will create a second variable to store this sequence
+  pcr_sequence <- paste0("5; ",
+                         whole_pcr_sequence,
+                         " 3'")
+```
+
+For the next block of code we will begin counting up each nucleotide within the annealing sequence and storing the individual counts of each base. We must account for all plausible bases within the sequence. Not only will we counting the typical A's, T's, C's, and G's, but we need to count up all the ambiguous bases too.
+
+![Ambiguous base chart]({{ site.url }}/assets/pcr-primer-calculator-using-r-ambiguous-chart.jpg)
+[Source](https://www.dnabaser.com/articles/IUPAC%20ambiguity%20codes.html)
+
+```r
+  # get a count of each nucleotide within the sequence
+  g_cnt <- stringr::str_count(annealing_seq, "G")
+  c_cnt <- stringr::str_count(annealing_seq, "C")
+  t_cnt <- stringr::str_count(annealing_seq, "T")
+  a_cnt <- stringr::str_count(annealing_seq, "A")
+
+  # get a count of each ambiguous base within the sequence
```

Comment:
I wonder if there is a way to just count any letter that isn't G, C, T or A, without having to count each one individually?